### ISSUE TITLE
Some Unicode symbols get rearranged in textboxes

### ISSUE THREAD

**Issue by HactarCE** [awesomeWM/awesome#2720]
Output of `awesome --version`:

```
awesome v4.3 (Too long)
 • Compiled against Lua 5.3.5 (running with Lua 5.3)
 • D-Bus support: ✔
 • execinfo support: ✔
 • xcb-randr version: 1.6
 • LGI version: 0.9.2
```

**How to reproduce the issue:**

1. Install a [Nerd Font](http://nerdfonts.com/) (or possibly some other similar symbols font; adjust examples accordingly)
2. Display one textbox with two symbols from symbol font:
    * `wibox.widgets.textbox("<span font='Symbols Nerd Font'>\u{fc65}\u{fa7d}</span>")`
3. Display another textbox with the same two symbols in reverse order:
    * `wibox.widgets.textbox("<span font='Symbols Nerd Font'>\u{fa7d}\u{fc65}</span>")`

**Actual result:**

In both textboxes, the volume symbol is displayed before the alert symbol, despite being listed in a different order in each textbox

**Expected result:**

In the first textbox, the alert symbol should be displayed first, and then the volume symbol. In the second textbox, the volume symbol should be displayed first, and then the alert symbol.

**Further information:**

The textboxes act as normal if the markup string begins with a character such as `a`. Thus one can prepend `<span size='0'>a</span>` to a string as a workaround.

**Comment by psychon** [comment]
This is also reproducible without the font:
![foo](https://user-images.githubusercontent.com/89482/53902159-e4aae380-4040-11e9-8530-5420792a0f32.png)

However, this second symbol seems... interesting. It seems to be `\u{fc65}` if I read the text for the other symbol correctly. Googling for U+FC65 finds a page saying that [this symbol is Arabic Ligature Yeh with Hamza Above with Zain Final Form](https://www.compart.com/en/unicode/U+FC65). "Arabic" makes me think you found Pango's right-to-left support.

If I could figure out how to produce these symbols in a terminal, I could try reproducing this problem with `pango-view --font "Symbols Nerd Font" --text "This is the text to display"`. I bet that this also has the same order behaviour that you are seeing in AwesomeWM. Feel free to try this out and report a bug against Pango with it, but personally I expect that they would explain right-to-left in response to such a bug report.

An alternative might be to report a bug against the font. Somehow abusing arabic characters for something else feels wrong to me.

P.S.: U+FA7D  also feels like it is something else: https://www.compart.com/en/unicode/U+FA7D

**Comment by psychon** [comment]
Found it (I can just copy&paste from the web pages I linked to above):
```
pango-view --font "Symbols Nerd Font" --text "墳ﱥ"
pango-view --font "Symbols Nerd Font" --text "ﱥ墳"
```
(I can also reproduce the problem not appearing if I add an "a" in front of the text)
So this seems to indeed be Pango detecting that that Arabic character is right-to-left text.

**Comment by psychon** [comment]
Ugly hack that no one should ever use and that might break when upgrading to newer versions of AwesomeWM (this hack basically disables Pango's "guess the direction"):
```lua
    local function my_textbox(...)
        local tb = wibox.widget.textbox(...)
        tb._private.layout:set_auto_dir(false)
        tb._private.ctx:set_base_dir(require("lgi").Pango.Direction.LTR)
        return tb
    end
```
(Then use `my_textbox` instead of `wibox.widget.textbox` when setting up the widgets with RTL characters.)

**Comment by Ryuno-Ki** [comment]
@psychon Tried <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>Ufa7d</kbd> in terminal?

See https://superuser.com/questions/59418/how-to-type-special-characters-in-linux

**Comment by HactarCE** [comment]
Thanks for the response/investigation! Automatic right-to-left detection makes sense here, so I don't think I'll file a Pango bug. :+1:

**Comment by ahauser31** [comment]
I know this is closed and all, but I ran into the same problem with the nerd fonts. By googling I found that there is a control character to set text left-to-right (and another one for right-to-left). So, to tell Pango that the text that follows is left-to-right even though it may have "arabic" (nerd font) symbols, I place `&#x200E;` (character U+200E) in my markup at the beginning and the problem is solved. Only issue is that the `space` character does not work, I assume it's not present in arabic. But that I can work around.
